src/charm.py

```diff
@@ -131,60 +130,43 @@ def _on_reset_instance_action(self, event: ActionEvent) -> None:
                 # Otherwise PostgreSQL will prevent it if there are open connections.
                 db_client = DatabaseClient(datasource=datasource, alternative_database="template1")
                 db_client.erase()
-            self._synapse.execute_migrate_config(container)
+            synapse.execute_migrate_config(container=container, charm_state=self._charm_state)
             logger.info("Start Synapse database")
             self.pebble_service.replan(container)
             results["reset-instance"] = True
-        except (psycopg2.Error, ops.pebble.PathError, CommandMigrateConfigError) as exc:
+        except (psycopg2.Error, ops.pebble.PathError, synapse.CommandMigrateConfigError) as exc:
             self.model.unit.status = ops.BlockedStatus(str(exc))
             event.fail(str(exc))
             return
         # results is a dict and set_results expects _SerializedData
         event.set_results(results)  # type: ignore[arg-type]
         self.model.unit.status = ops.ActiveStatus()
 
-    def _get_random_password(self) -> str:
-        """Get random password. Extracted from postgresql-k8s charm.
-
-        Returns:
-            random password.
-        """
-        choices = string.ascii_letters + string.digits
-        password = "".join([secrets.choice(choices) for i in range(16)])
-        return password
-
     def _on_register_user_action(self, event: ActionEvent) -> None:
         """Reset instance and report action result.
 
         Args:
             event: Event triggering the reset instance action.
         """
-        results = {"register-user": False, "user-password": ""}
         container = self.unit.get_container(SYNAPSE_CONTAINER_NAME)
         if not container.can_connect():
             self.unit.status = ops.MaintenanceStatus("Waiting for pebble")
             return
-        registration_shared_secret = self._synapse.get_configuration_field(
-            container=container, fieldname="registration_shared_secret"
-        )
+        registration_shared_secret = synapse.get_registration_shared_secret(container=container)
         if registration_shared_secret is None:
-            event.fail("registration_shared_secret was not found")
+            event.fail("registration_shared_secret was not found, please check the logs")
             return
         user_data = {
             "username": event.params["username"],
             "admin": event.params["admin"],
-            "password": self._get_random_password(),
         }
-        user = User(**user_data)
         try:
-            synapse_api.register_user(
-                registration_shared_secret=registration_shared_secret, user=user
-            )
-            results["register-user"] = True
-            results["user-password"] = user.password
-        except synapse_api.SynapseAPIError as exc:
+            user = User(**user_data)
```

Comment:
rather than dict unpacking, can't we just pass the event data directly? E.g., `User(username=event.params["username"], "admin"=event.params["admin"])`, one fewer level of indirection plus we get linting checks on the arguments

Reply:
No problem, changed.